Source/WebKit/Shared/WebCoreArgumentCoders.cpp

```diff
@@ -420,11 +312,10 @@ std::optional<DOMCacheEngine::Record> ArgumentCoder<DOMCacheEngine::Record>::dec
         return std::nullopt;
 
     if (hasSharedBufferBody) {
-        RefPtr<SharedBuffer> buffer;
-        if (!decodeSharedBuffer(decoder, buffer))
+        if (auto buffer = decoder.decode<RefPtr<SharedBuffer>>())
+            responseBody = buffer->releaseNonNull();
```

Comment:
.. and here you are decoding a refptr
(source of your bug that causes the asserts in tests)

This is the reason why it would be better to do the encoder in consistent manner, see above.